### PR TITLE
Fix hang at unmount when Ctrl-C is pressed

### PR DESCRIPTION
**Pull request by CrendKing**
More details at https://github.com/kopia/kopia/issues/716

### CODE REVIEW

**Comment by jkowalski** [comment]
Calling `ctrl.Done()` without reading from it has no effect, it will simply return a channel without performing any operation.

The sequence should be:

when ctrl-C is pressed we call `ctrl.Unmount()`
`ctrl.Unmount()` should eventually/asynchronously close channel returned by `ctrl.Done()` channel, somehow this is not happening.

I think `mount_webdav.go` is missing close(c.done) inside `Unmount()` or you can do something like this in line 59:

```
done := make(chan struct{})

srv.RegisterOnShutdown(func() {
		close(done)
	})
```

If you can test whether this works on Windows, and update the PR, I'll be happy to merge it.

**Comment by CrendKing** [comment]
Yes, it works. Updated the PR. I also kept the hoisted common code in cli/command_mount.go, but I'm not 100% sure if that has no side effect. Also not sure if it is safe to remove the empty `<-ctrl.Done()` case. If you prefer, I can restore that file altogether.

File: cli/command_mount.go [hunk 1]
```diff
@@ -86,13 +86,12 @@ func runMountCommand(ctx context.Context, rep repo.Repository) error {
...
 
+		// Reporting clean unmount in case of interrupt signal.
+		<-ctrl.Done()
+
 	case <-ctrl.Done():
-		log(ctx).Infof("Unmounted.")
-		return nil
 	}
 
-	// Reporting clean unmount in case of interrupt signal.
```
Comment (jkowalski) [inline cli/command_mount.go:94]: this entire file should be reverted.

Comment (CrendKing) [reply cli/command_mount.go]: Done. Sorry about that.

**Comment by jkowalski** [comment]
thank you!

**Pull request merged**